tests/data_engine/test_querying.py

```diff
@@ -27,6 +28,57 @@ def test_simple_filter(ds):
     assert q.to_dict() == expected
 
 
+def test_versioning_query_ts_format(ds):
+    add_int_fields(ds, "x")
+
+    # timestamp
+    ds2 = ds[ds[Field("x", as_of_time=1700604000)] > 1]
+    q = ds2.get_query()
+    assert q.to_dict() == {'gt': {'data': {'as_of': 1700604000, 'field': 'x', 'value': 1}}}
+
+    # datetime
+    ds2 = ds[ds[Field("x", as_of_time=dateutil.parser.parse("Wed 22 Nov 2023"))] > 1]
+    q = ds2.get_query()
+    assert q.to_dict() == {'gt': {'data': {'as_of': int(dateutil.parser.parse("Wed 22 Nov 2023").timestamp()),
+                                           'field': 'x', 'value': 1}}}
```

Comment:
Separate these into two different tests.
This way if datetime serialization fails, you see X on the datetime test, but not on the timestamp one